### PR TITLE
Add support for alpha.36 changes

### PR DESCRIPTION
**Pull request by cmelchior**
Looks like https://github.com/realm/ros/issues/360 is causing a lot of problems

### CODE REVIEW

File: realm/realm-library/src/objectServer/java/io/realm/internal/network/OkHttpAuthenticationServer.java [hunk 1]
```diff
@@ -38,7 +38,7 @@
     public static final MediaType JSON = MediaType.parse("application/json; charset=utf-8");
     private static final String ACTION_LOGOUT = "revoke"; // Auth end point for logging out users
     private static final String ACTION_CHANGE_PASSWORD = "password"; // Auth end point for changing passwords
-    private static final String ACTION_LOOKUP_USER_ID = "/users/:provider:/:providerId:"; // Auth end point for looking up user id
+    private static final String ACTION_LOOKUP_USER_ID = "users/:provider:/:providerId:"; // Auth end point for looking up user id
```
Comment (nhachicha) [inline realm/realm-library/src/objectServer/java/io/realm/internal/network/OkHttpAuthenticationServer.java]: why removing the leading `/` are you aligning URI?

File: realm/realm-library/src/objectServer/java/io/realm/internal/network/LookupUserIdResponse.java [hunk 2]
```diff
@@ -127,21 +133,21 @@ public boolean isAdmin() {
...
     public Map<String, String> getMetadata() { return metadata; }
 
-    private static Map<String, String> jsonToMap(JSONObject json) throws JSONException {
-        Map<String, String> map = new HashMap<>();
-        if(json != JSONObject.NULL) {
-            map = toMap(json);
-        }
-        return map;
-    }
+    public Map<String, String> getAccounts() { return accounts; }
```
Comment (nhachicha) [inline realm/realm-library/src/objectServer/java/io/realm/internal/network/LookupUserIdResponse.java]: any tests that covers `getAccounts ` (and by extension `jsonToMap`)

File: realm/realm-library/src/objectServer/java/io/realm/internal/network/OkHttpAuthenticationServer.java (hunk 1 above)
Comment (cmelchior) [reply realm/realm-library/src/objectServer/java/io/realm/internal/network/OkHttpAuthenticationServer.java]: Yes, the first part of the URL is normalized so it always ends with `/`

**Comment by cmelchior** [comment]
Ready for review @realm/java

File: realm/realm-library/src/objectServer/java/io/realm/internal/network/LookupUserIdResponse.java (hunk 2 above)
Comment (cmelchior) [reply realm/realm-library/src/objectServer/java/io/realm/internal/network/LookupUserIdResponse.java]: Yes: https://github.com/realm/realm-java/pull/5313/files#diff-5ad3dd1db70a9dde54cfd2af71abbbe7R625

File: realm/realm-library/src/syncIntegrationTest/java/io/realm/SyncedRealmTests.java [hunk 3]
```diff
@@ -44,7 +44,7 @@
  * Catch all class for tests that not naturally fit anywhere else.
  */
 @RunWith(AndroidJUnit4.class)
-@Ignore("See https://github.com/realm/realm-java/issues/5177. All waitForInitialRemoteData tests seem to fail. Must be fixed")
+@Ignore("See https://github.com/realm/ros/issues/360")
```
Comment (beeender) [inline realm/realm-library/src/syncIntegrationTest/java/io/realm/SyncedRealmTests.java]: not released?

**Pull request merged**